test/3x/torch/quantization/weight_only/test_gptq.py

```diff
@@ -1,295 +1,183 @@
-
-
-class GPTQLLMDataLoaderDict(GPTQLLMDataLoader):
-    def __iter__(self):
-        for i in range(10):
-            yield {
-                "input_ids": torch.ones([1, self.length], dtype=torch.long),
-                "attention_mask": torch.ones([1, self.length], dtype=torch.long),
-            }
-
-
-from tqdm import tqdm
-
-from neural_compressor.torch.algorithms.weight_only.gptq import move_input_to_device
-
-
-def run_fn_for_gptq(model, dataloader_for_calibration, *args):
-    logger.info("Collecting calibration inputs...")
-    for batch in tqdm(dataloader_for_calibration):
-        batch = move_input_to_device(batch, device=None)
-        try:
-            if isinstance(batch, tuple) or isinstance(batch, list):
-                model(batch[0])
-            elif isinstance(batch, dict):
-                model(**batch)
-            else:
-                model(batch)
-        except ValueError:
-            pass
-    return
-
-
-class TestGPTQ(unittest.TestCase):
-    @classmethod
-    def setUpClass(self):
-        pass
+class TestGPTQQuant:
+    def setup_class(self):
+        self.tiny_gptj = transformers.AutoModelForCausalLM.from_pretrained(
+            "hf-internal-testing/tiny-random-GPTJForCausalLM",
+        )
+        self.example_inputs = torch.tensor([[10, 20, 30, 40, 50, 60]], dtype=torch.long)
+        # record label for comparison
+        self.label = self.tiny_gptj(self.example_inputs)[0]
 
-    @classmethod
-    def tearDownClass(self):
+    def teardown_class(self):
         pass
 
-    def setUp(self):
-        # print the test name
-        logger.info(f"Running TestGPTQ test: {self.id()}")
-
-    def test_gptq(self):
-        # Ported from test/adaptor/pytorch_adaptor/test_weight_only_adaptor.py
-        # TestPytorchWeightOnlyAdaptor.test_GPTQ_fixed_length_quant
-
-        dataloader = GPTQLLMDataLoader()
-
-        # case 1: tensor
-        model = get_gpt_j()
-        input = torch.ones([1, 512], dtype=torch.long)
-        out0 = model(input)
-        model_1 = copy.deepcopy(model)
-        device = None
-        from neural_compressor.torch.algorithms.weight_only.gptq import DataloaderPreprocessor
-
-        dataloaderPreprocessor = DataloaderPreprocessor(
-            dataloader_original=dataloader, use_max_length=False, pad_max_length=512, nsamples=128
-        )
-        dataloader_for_calibration = dataloaderPreprocessor.get_prepared_dataloader()
-
+    def test_accuracy_improvement(self):
+        # test_default_rtn_config
+        model = copy.deepcopy(self.tiny_gptj)
+        quant_config = get_default_rtn_config()
+        model = quantize(model, quant_config, run_fn=run_fn)
+        rtn_label = model(self.example_inputs)[0]
+        rtn_atol = (rtn_label - self.label).amax()
+        # test_default_gptq_config
+        model = copy.deepcopy(self.tiny_gptj)
+        quant_config = get_default_gptq_config()
+        model = quantize(model, quant_config, run_fn=run_fn)
```

Comment:
I'm a little confused about using GPTQ with a custom dataloader, can you please provide an end-to-end example?

Reply:
```
dataloder = Dataloader()
def run_fn(model):
    for data, label in dataloader:
        model(data)
```
The dataloader is defined/imported in the same file as run_fn then python can detect it.